src/fuzzfetch/fetch.py

```diff
@@ -457,12 +457,12 @@ def __init__(self, target, branch, build, flags, platform=None, nearest=None):
                 # If start date is outside the range of the newest/oldest available build, adjust it
                 if asc:
                     start = max(start, now - timedelta(days=364))
-                    end = now
+                    end = min(start, now)
                 else:
-                    start = min(start, now)
                     end = now - timedelta(days=364)
+                    start = max(min(start, now), end)
```

Comment:
Ditto here. `end` is set to `now - 364 days` right above, so `max(now,end)` is always going to return `now`, so this change has no effect.

Reply:
This one however, I think is correct.  Min(start, now) ensures we have the earliest date, but then max(start, end) ensures that our starting date is greater than the end date.

Reply:
Yep .. you're right. What I said about `max(now,end)` always being `now` is true, but if `start` is in the future, then the `min()` will not be a no-op.